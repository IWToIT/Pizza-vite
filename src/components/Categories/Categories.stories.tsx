import type { Meta, StoryObj } from "@storybook/react";
import { Categories } from "./index";

const meta: Meta<typeof Categories> = {
  component: Categories,
};

export default meta;
type Story = StoryObj<typeof Categories>;

export const Primary: Story = {
  args: {
    onChangeCategory: (index: number) => {},
    value: 0,
  },
};
