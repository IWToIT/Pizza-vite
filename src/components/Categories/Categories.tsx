import React from 'react';
import { categories } from '@/utils/consts';

export type CategoriesProps = {
  value: number;
  onChangeCategory: (index: number) => void;
};

const Categories: React.FC<CategoriesProps> = React.memo(({ value, onChangeCategory }) => {
  return (
    <div className="categories" data-testid="cat-element">
      <ul className="categories__list">
        {categories.map((categoryName, i) => (
          <li
            key={i}
            onClick={() => onChangeCategory(i)}
            className={`categories__list-item${value === i ? '--active' : ''}`}
          >
            {categoryName}
          </li>
        ))}
      </ul>
    </div>
  );
});

export default Categories;
