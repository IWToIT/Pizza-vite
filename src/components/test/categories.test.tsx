import React from "react";
import Categories from "../Categories/Categories";
import { render } from "./test-utils";
import { screen, within } from "@testing-library/react";

test("highlighting the active category", () => {
  const value = 2;
  const onChangeCategory = jest.fn();
  render(<Categories value={value} onChangeCategory={onChangeCategory} />);
  expect(screen.getByTestId("cat-element")).toBeInTheDocument();
  expect(screen.getByTestId("cat-element")).toMatchSnapshot();
  const categoryList = screen.getByRole("list");
  const categoryItems = within(categoryList).getAllByRole("listitem");
  expect(categoryItems[2]).toHaveClass("active");
});
