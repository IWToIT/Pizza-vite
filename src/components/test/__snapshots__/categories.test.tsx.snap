// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlighting the active category 1`] = `
<div
  class="categories"
  data-testid="cat-element"
>
  <ul>
    <li
      class=""
    >
      Все
    </li>
    <li
      class=""
    >
      Мясные
    </li>
    <li
      class="active"
    >
      Вегетарианская
    </li>
    <li
      class=""
    >
      Гриль
    </li>
    <li
      class=""
    >
      Острые
    </li>
    <li
      class=""
    >
      Закрытые
    </li>
  </ul>
</div>
`;

exports[`выделение активной категории 1`] = `
<div
  class="categories"
  data-testid="cat-element"
>
  <ul>
    <li
      class=""
    >
      Все
    </li>
    <li
      class=""
    >
      Мясные
    </li>
    <li
      class="active"
    >
      Вегетарианская
    </li>
    <li
      class=""
    >
      Гриль
    </li>
    <li
      class=""
    >
      Острые
    </li>
    <li
      class=""
    >
      Закрытые
    </li>
  </ul>
</div>
`;
